src/stack.ts

```diff
@@ -17,8 +17,10 @@ import * as pulumi from '@pulumi/pulumi';
 import { debug } from '@pulumi/pulumi/log';
 import { StackComponentResource, StackOptions } from './types';
 import { AppConverter, StackConverter } from './converters/app-converter';
+import { PulumiSynthesizer } from './synthesizer';
+import { CdkConstruct } from './interop';
 
-class StackComponent extends pulumi.ComponentResource implements StackComponentResource {
```

Comment:
Nit: not very informed about the trade-offs here but at the very abstract philosophical level I dread deep sub-classing and preferred the previous form, which I understand might lead to some code duplication but perhaps we'd rather deal with that then a deeper hierarchy. Having this hierarchy in Pulumi proper (Resource -> ComponentResource) already created some formidable difficulties, the team wishes Resource itself was an interface.

Reply:
I'll turn this into an interface. It primarily exists to avoid circular dependencies.